stdlib/regex.gr

```diff
@@ -3954,11 +3954,12 @@ let regexReplaceHelp =
   let mut out = []
   let rec loop = searchPos => {
     let state = Array.make(rx.reNumGroups, None)
+    let inStart = max(0, searchPos - rx.reMaxLookbehind)
     let poss = searchMatch(
       rx,
       buf,
       searchPos,
-      searchPos,
```

Comment:
This was breaking the anchoring behavior, since the "am I at the start of the input?" check always succeeded due to us saying that the input started at `searchPos` (the current position). For the regular expression `^.`, the lookbehind is `1`, so this will be `0` when `searchPos == 0` and `searchPos - 1` for every other position.